cmd/acra_genauth/acra_genauth.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(file, crypted, 0644)
+}
+
+func (hp HashedPasswords) SetPassword(name, password string) (err error) {
+	if len(password) == 0 {
+		return errors.New("passwords is empty")
+	}
+	salt := cmd.RandomStringBytes(16)
+	argon2Params := cmd.InitArgon2Params()
+	hashBytes, err := cmd.HashArgon2(password, salt, argon2Params)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	a := cmd.UserAuth{Salt: salt, Hash: hashBytes, Argon2Params: argon2Params}
+	hp[name] = a.UserAuthString(":", ",")
+	return nil
+}
+
+func ParseHtpasswdFile(file string, keystore *keystore.FilesystemKeyStore) (passwords HashedPasswords, err error) {
+	htpasswdBytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return
+	}
+	key, err := keystore.GetAuthKey(false)
+	if err != nil {
+		return
+	}
+	SecureCell := cell.New(key, cell.CELL_MODE_SEAL)
+	authData, err := SecureCell.Unprotect(htpasswdBytes, nil, nil)
+	if err != nil {
+		return
+	}
+	return ParseHtpasswd(authData)
+}
+
+func ParseHtpasswd(htpasswdBytes []byte) (passwords HashedPasswords, err error) {
+	authFieldsCount := 4
+	lines := strings.Split(string(htpasswdBytes), LineSeparator)
+	passwords = make(map[string]string)
+	for index, line := range lines {
+		line = strings.Trim(line, " ")
+		if len(line) == 0 {
+			continue
+		}
+		parts := strings.Split(line, PasswordSeparator)
+		if len(parts) != authFieldsCount {
+			err = errors.New(fmt.Sprintf("wrong line no. %d, unexpected number (%v) of splitted parts split by %v", index+1, len(parts), PasswordSeparator))
+			return
+		}
+		for i, part := range parts {
+			parts[i] = strings.Trim(part, " ")
+		}
+		_, alreadyExists := passwords[parts[0]]
+		if alreadyExists {
+			err = errors.New(fmt.Sprintf("wrong line no. %d, user (%v) already defined", index, parts[0]))
+			return
+		}
+		passwords[parts[0]] = strings.Join(parts[1:authFieldsCount], PasswordSeparator)
+	}
+	return
+}
+
+func RemoveUser(file, user string, keystore *keystore.FilesystemKeyStore) error {
+	passwords, err := ParseHtpasswdFile(file, keystore)
+	if err != nil {
+		return err
+	}
+	_, ok := passwords[user]
+	if !ok {
+		return errors.New("user not found in file")
+	}
+	delete(passwords, user)
+	return passwords.WriteToFile(file, keystore)
+}
+
+func SetPassword(file, name, password string, keystore *keystore.FilesystemKeyStore) error {
+	_, err := os.Stat(file)
+	passwords := HashedPasswords(map[string]string{})
+	if err == nil {
+		passwords, err = ParseHtpasswdFile(file, keystore)
+		if err != nil {
+			return err
+		}
+	}
+	err = passwords.SetPassword(name, password)
+	if err != nil {
+		return err
+	}
+	return passwords.WriteToFile(file, keystore)
+}
+
+func main() {
+	set := flag.Bool("set", false, "Add/update password for user")
+	remove := flag.Bool("remove", false, "Remove user")
+	user := flag.String("user", "", "User")
+	pwd := flag.String("pwd", "", "Password")
```

Comment:
lets use full word as parameter (`pwd` -> `password`) if we use full versions of words in other params or use short everywhere (rm|set|usr|pwd|etc)

Reply:
long!